examples/third_party_integration/flux/flux_test.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flux
+
+import (
+	"context"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"os"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+	"sigs.k8s.io/e2e-framework/third_party/flux"
+	"testing"
+	"time"
+)
+
+var curDir, _ = os.Getwd()
+
+func TestFluxRepoWorkflow(t *testing.T) {
+	gitRepoName := "hello-world"
+	kustomizationName := "hello-world"
+
+	feature := features.New("Install resources by flux").
+		Setup(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			manager := flux.New(c.KubeconfigFile())
+			err := manager.CreateGitRepo(gitRepoName, "https://github.com/matrus2/go-hello-world", flux.WithBranch("main"))
+			if err != nil {
+				t.Fatal("failed to create git repository", err)
+			}
+			// Set --prune so that once deleteKustomization will be called all corresponding resources will be removed
+			err = manager.CreateKustomization(kustomizationName, "GitRepository/hello-world.flux-system", flux.WithPath("template"), flux.WithArgs("--target-namespace", c.Namespace(), "--prune"))
```

Comment:
Question: where is the kustomization file `GitRepository/hello-world.flux-system` coming from ? Is it pulled from the repo?

Reply:
GitRepository/hello-world.flux-system is a reference to GitRepository object in the flux-system namespace created in previous step. Basically, GitRepository is the declaration of a git repository and reference to the source of manifest files: 
```yaml
  apiVersion: source.toolkit.fluxcd.io/v1
  kind: GitRepository
  metadata:
    name: hello-world
    namespace: flux-system
  spec:
    interval: 10m
    ref:
      branch: main
    url: ssh://git@github.com/matrus2/go-hello-world
```
The Kustomization is a specification, which files form Git Repository and how they should be deployed. In this example we deploy files from path `template` within 1 minute interval to target namespace default. See below the example:   
```yaml
  apiVersion: kustomize.toolkit.fluxcd.io/v1
  kind: Kustomization
  metadata:
    name: hello-world
    namespace: flux-system
  spec:
    interval: 1m
    targetNamespace: default
    path: template
    prune: true
    sourceRef:
      kind: GitRepository
      name: hello-world
  ```